### PR TITLE
CLOUDP-162861: Remove unnnecessary new constructors from the auth pkg

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-162861

Trying to clean up auth login/register and setup is becoming too complected due to a big amount of technical debt and inter-dependency, starting small with a small clean up of unnecessary methods that expose a lot of the issues I'm seeing

I'll pause trying to fix the unit tests deleting config files after this PR since it's becoming bigger and I have other tickets to look at

### CODE REVIEW

File: internal/cli/atlas/setup/setup_cmd.go [hunk 1]
```diff
@@ -83,7 +83,7 @@ This command will help you
 	return opts.quickstart.Run()
 }
 
-func (opts *Opts) PreRun(ctx context.Context) error {
```
Comment (gssbzn) [inline internal/cli/atlas/setup/setup_cmd.go:86]: never returns an error

File: internal/cli/atlas/setup/setup_cmd.go [hunk 2]
```diff
@@ -129,8 +127,8 @@ func Builder() *cobra.Command {
 	qsOpts.LabelValue = labelValue
 	opts := &Opts{
 		register:   auth.NewRegisterFlow(loginOpts),
-		login:      auth.NewLoginFlow(loginOpts),
-		quickstart: quickstart.NewQuickstartFlow(qsOpts),
+		login:      loginOpts,
+		quickstart: qsOpts,
```
Comment (gssbzn) [inline internal/cli/atlas/setup/setup_cmd.go:131]: quick-start has 2 logins, bringing the count o 3

File: internal/cli/atlas/setup/setup_cmd_test.go [hunk 3]
```diff
@@ -168,7 +168,8 @@ func Test_setupOpts_RunSkipRegister(t *testing.T) {
 		Return(nil).
 		Times(1)
 
-	require.NoError(t, opts.PreRun(ctx))
+	// FIXME: CLOUDP-162860 side effect dependant test
```
Comment (gssbzn) [inline internal/cli/atlas/setup/setup_cmd_test.go:171]: this test is for run but somehow depends on prerun having been called

File: internal/cli/auth/login.go [hunk 4]
```diff
@@ -48,12 +48,12 @@ type LoginConfig interface {
 }
 
 const (
-	AlreadyAuthenticatedError      = "you are already authenticated with an API key (Public key: %s)"
+	alreadyAuthenticatedError      = "you are already authenticated with an API key (Public key: %s)"
```
Comment (gssbzn) [inline internal/cli/auth/login.go:51]: reducing visibility, I have a follow up task to clean up these const seems they are only used in 1 place but it requires some changes to tests

**Review by wtrocki — APPROVED** [review]

**Pull request merged**